library/ssl_tls.c

```diff
@@ -8619,18 +8648,24 @@ int mbedtls_ssl_setup( mbedtls_ssl_context *ssl,
     /* Set to NULL in case of an error condition */
     ssl->out_buf = NULL;
 
-    ssl->in_buf = mbedtls_calloc( 1, MBEDTLS_SSL_IN_BUFFER_LEN );
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    ssl->in_buf_len = in_buf_len;
+#endif
+    ssl->in_buf = mbedtls_calloc( 1, in_buf_len );
     if( ssl->in_buf == NULL )
     {
-        MBEDTLS_SSL_DEBUG_MSG( 1, ( "alloc(%d bytes) failed", MBEDTLS_SSL_IN_BUFFER_LEN) );
+        MBEDTLS_SSL_DEBUG_MSG( 1, ( "alloc(%d bytes) failed", in_buf_len) );
         ret = MBEDTLS_ERR_SSL_ALLOC_FAILED;
         goto error;
     }
 
-    ssl->out_buf = mbedtls_calloc( 1, MBEDTLS_SSL_OUT_BUFFER_LEN );
+#if defined(MBEDTLS_SSL_VARIABLE_BUFFER_LENGTH)
+    ssl->out_buf_len = MBEDTLS_SSL_OUT_BUFFER_LEN;
```

Comment:
Could we use `out_buf_len` here instead of `MBEDTLS_SSL_OUT_BUFFER_LEN`? This would help us avoid repeating ourselves, since we already said `out_buf_len` is equal to `MBEDTLS_SSL_OUT_BUFFER_LEN`. This makes maintenance easier since we only need to update one spot in code, not two, and there is less of a chance that we forget to update both values.

Reply:
This looks like the only place where `ssl->out_buf_len` is set. Are we expecting any users to ever modify this directly? If we want to expressly disallow that, we should add a comment to the structure definition given that the struct is in a public API header.

Reply:
Agreed, `out_buf_len` should be used, I'll fix that.

I don't see much point in merging this into development until its possible to actually do something with it. There's a lot of tests that need to be added first before we get to that point though.